airflow/datasets/__init__.py

```diff
@@ -42,3 +43,13 @@ def _check_uri(self, attr, uri: str):
         parsed = urlsplit(uri)
         if parsed.scheme and parsed.scheme.lower() == "airflow":
             raise ValueError(f"{attr.name!r} scheme `airflow` is reserved")
+
+
+class any_of(UserList):
+    """
+    Custom collection to represent a list of Datasets that will trigger a
+    DAG run when any of them is updated.
+    """
+
+    def __init__(self, *datasets: Dataset):
+        super().__init__([*datasets])
```

Comment:
I don’t see how this can’t simply be

```python
class anyof(list):
    pass
```

Reply:
Looking at it firther I don't think implementing this as a class makes sense. It isn't equivalent to a Dataset. We are coercing the rule now into Dataset-like thing which isnt sound imho.

Reply:
@uranusjr if this was a subclass of list you would need to use it like this:

```python
any_of([dataset1, dataset2])

# with the actual implementation you do:

any_of(dataset1, dataset2)
```

@bolkedebruin can you suggest a better place for it? I thought it was handy to import both `Dataset` and `any_of` from the same module as they are closely related.

Reply:
I think it is confusing as one is a rule and the other is a Data class. Ie. one references something while the other operates on the one that references.

I think this should be located in rules.py